src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,240 @@
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        boolean isDone;
+        while (s.hasNext()) {
+            Task task = null;
+            //System.out.println(s.nextLine());
+            String line = s.nextLine();
+            String words[] = line.split(" \\| ");
+            String firstCommand= words[0];
+            isDone= words[1].equals("1");
+
+            switch(firstCommand){
+            case "T":
+                task = new Todo(words[2],true);
+                break;
+            case "D":
+                task = new Deadline(words[2],words[3], true);
+                break;
+            case "E":
+                task = new Event(words[2],words[3], true);
+                break;
+
+            }
+            if(task!=null) {
+                task.isDone=isDone;
+                addNewTask(task);
+            }
+        }
+    }
+
+    
+    public static void addNewTask(Task task) {
+        tasks.add(task);
+        taskCounter++;
+    }
+
+
+    private static void updateFile() throws IOException {
+        writeToFile(fileDir, "");
+        if(taskCounter >0) {
+            for (int i = 0; i < taskCounter; i++) {
+                appendToFile(fileDir, tasks.get(i).toStringFile());
+            }
+        }
+
+    }
+
+
+    public static void createFileHierarchy(String folderPath, String filePath) throws IOException {
+        File folder = new File(folderPath);
+        File file = new File(filePath);
+        if(folder.exists()) {
+            if(file.exists()){
+                processFile(filePath);
+                return;
+            }
+            else{
+                file.createNewFile();
+            }
+        }
+        else {
+            folder.mkdir();
+            file.createNewFile();
+        }
+        
+    }
+
+
+    public static void printList() {
+        printDivider();
+        if(taskCounter >0) {
+            printLn("Here are the tasks in your list:");
+
+            for (int i = 0; i < taskCounter; i++) {
+                printLn((i + 1) +"."+ tasks.get(i).toString());
+            }
+        }
+        else {
+            printLn("Nothing added!");
+        }
+        printDivider();
+    }
+
+
+    public static void printStartMsg() {
+        String logo = indent + "    .___     __\n" +
+                indent + "  __| _/_ __|  | __ ____\n" +
+                indent + " / __ |  |  \\  |/ // __ \\\n" +
+                indent + "/ /_/ |  |  /    <\\  ___/\n" +
+                indent + "\\____ |____/|__|_ \\\\___  >\n" +
+                indent + "     \\/          \\/    \\/";
+        printLn("Hello from\n" + logo);
+        printDivider();
+        printLn("Hello! I'm Duke\n" +
+                indent + "What can I do for you?");
+        printDivider();
+    }
+
+
+    public static void checkValidIndex(String word, String command) {
```

Comment:
As the method name is `checkValidIndex`, I would not expect the command to be executed in this method